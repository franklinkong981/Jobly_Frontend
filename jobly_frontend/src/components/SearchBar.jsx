import React, {useState} from "react";
import {useFormik} from "formik";


function SearchBar({filterFunc, placeholder}) {
  const formik = useFormik({
    initialValues: {
      query: ''
    },
    validateOnChange: false,
    validateOnBlur: false,
    async onSubmit(values) {
      await filterFunc(values.query);
    }
  });


  return (
    <section className="col-md-4">
      <form className="SearchBar" onSubmit={formik.handleSubmit}>
        <input id="SearchBar-query-field" className="SearchBar-input" type="text" name="query"
        size="25" value={formik.values.query} placeholder={placeholder} onChange={formik.handleChange}/>

        <button className="SearchBar-search-button" type="submit">Search</button>
      </form>
    </section>
  );
}

export default SearchBar;