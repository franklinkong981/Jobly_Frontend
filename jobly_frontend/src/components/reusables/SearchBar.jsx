import React, {useState} from "react";
import "./SearchBar.css";

/**
 * Search bar that will be displayed at the top of the companies list and jobs list pages.
 * Users can type in a query and submit the form, then only the cmompanies/jobs whose names contain the query will be displayed.
 * 
 * To do this, once the form is submitted, the filterFunc function prop is called to filter the companies/jobs. 
 * This component is present in the CompanyList and JobList components.
 */
function SearchBar({filterFunc, placeholder}) {
  const [searchQuery, setSearchQuery] = useState("");

  function handleSubmit(evt) {
    evt.preventDefault();
    //This check prevents searching for a query that is just empty spaces.
    searchQuery.trim() ? filterFunc(searchQuery.trim()) : filterFunc();
    setSearchQuery(searchQuery => searchQuery.trim());
  }

  /** Used to update the value in the searchbar at any given time. */
  function handleChange(evt) {
    setSearchQuery(searchQuery => evt.target.value);
  }

  return (
    <div className="SearchBar mb-4">
      <form className="SearchBar-form form-inline" onSubmit={handleSubmit}>
        <input
          className="SearchBar-input form-control form-control-lg flex-grow-1"
          name="searchQuery"
          placeholder={placeholder}
          value={searchQuery}
          onChange={handleChange}
        />
        <button type="submit" className="SearchBar-submit btn btn-lg btn-primary">
          Search
        </button>
      </form>
    </div>
  );
}

export default SearchBar;