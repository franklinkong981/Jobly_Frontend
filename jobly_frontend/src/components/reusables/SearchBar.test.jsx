import React from "react";
import {render} from "@testing-library/react";

import SearchBar from "./SearchBar.jsx";
import { expect, it } from "vitest";

const testFilteringFunc = () => {
  return false;
}

it("renders without crashing", function() {
  render(<SearchBar filterFunc={testFilteringFunc} placeholder="This is for testing" />);
});

it("matches snapshot", function() {
  const {asFragment} = render(<SearchBar filterFunc={testFilteringFunc} placeholder="This is for testing" />);
  expect(asFragment()).toMatchSnapshot();
});