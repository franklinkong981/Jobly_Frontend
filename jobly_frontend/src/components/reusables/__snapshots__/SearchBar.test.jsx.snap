// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SearchBar mb-4"
  >
    <form
      class="SearchBar-form form-inline"
    >
      <input
        class="SearchBar-input form-control form-control-lg flex-grow-1"
        name="searchQuery"
        placeholder="This is for testing"
        value=""
      />
      <button
        class="SearchBar-submit btn btn-lg btn-primary"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
